### ISSUE TITLE
CME with Citizens 2 (calling async players causes ticking entity crash randomly)

### ISSUE THREAD

**Issue by Kainzo** [CitizensDev/Citizens2#508]
https://www.irccloud.com/pastebin/sD2PHILn/
http://pastebin.com/9feCD0r6
https://www.irccloud.com/pastebin/f62EE4xn/
http://pastebin.com/YuSi4ynq

https://hub.spigotmc.org/jira/browse/SPIGOT-1171?jql=

**Comment by fullwall** [comment]
What do you mean by 'calling async players'? I noticed that you are using an older build, which may be the cause: as far as I can tell in the latest build world.players is not touched at all.

**Comment by Soren025** [comment]
@fullwall Basically as a general explanation of what I observed after reviewing the nms code within the stack trace, there is a `List<HumanEntity>` being accessed in a non thread safe way in some form within the citizens plugin. As a result when `EntityLigntning` attempts to iterate that list it is throwing the `ConcurrentModificationException`. Spigot does not catch this and the server shuts down.

When this was happening the crash would occur about once every 5-6 hours. This is understandable given the nature of the issue.

It was also determined to be an issue caused by Citizens because the crashes only started once Citizens was added, and then no longer occurred once removed.

Granted we have no stack trace linking to Citizens, and for that matter it is quite impossible to produce such a thing. All I can offer is a stack trace identifying the list causing the `ConcurrentModificationException`, and the fact that Citizens was isolated to be the soul plugin that when added started the crashes, then once moved stopped them.
